File: app/src/main/java/br/com/ifood/tweetmood/data/rest/GoogleApi.java
package br.com.ifood.tweetmood.data.rest;

import br.com.ifood.tweetmood.BuildConfig;
import br.com.ifood.tweetmood.domain.model.Request.AnalyzeSentimentRequest;
import br.com.ifood.tweetmood.domain.model.Response.AnalyzeSentimentResponse;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

/**
 * Created by uchoa on 10/06/18.
 */

public interface GoogleApi {

    @POST(BuildConfig.API_VERSION + "documents:analyzeSentiment")
    Call<AnalyzeSentimentResponse> postAnalyzeSentiment(@Body AnalyzeSentimentRequest request);


}

File: app/src/main/kotlin/movile/marcus/com/br/moviletest/model/api/GoogleApi.kt
package movile.marcus.com.br.moviletest.model.api

import io.reactivex.Flowable
import movile.marcus.com.br.moviletest.BuildConfig
import movile.marcus.com.br.moviletest.model.data.SentimentResult
import okhttp3.RequestBody
import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.http.Query

interface GoogleApi {

    @POST("v1/documents:analyzeSentiment")
    fun analyzeTweetMood(@Body requestBody: RequestBody, @Query("key") key: String = BuildConfig.GOOGLE_NATURAL_LANGUAGE): Flowable<SentimentResult>
}